string.go

```diff
@@ -87,21 +88,23 @@ func formatHelper(l *State, fs string, argCount int) string {
 				buf := b.Bytes()
 				buf[len(buf)-1] = byte(c)
 			case 'i': // The fmt package doesn't support %i.
-				f = f[:len(f)-1] + "d"
+				f = f[:len(f)-1] + "u"
 				fallthrough
 			case 'd':
+				n := CheckInteger(l, arg)
```

Comment:
I think `CheckInteger` is wrong - it looks like it just calls `ToNumber` and casts to `int`. I think we should be doing something like:
```go
n := CheckNumber(l, arg)
ArgumentCheck(l, math.Floor(n) == n && -math.Pow(2, 63) <= n && n < math.Pow(2, 63), arg, "number has no integer representation")
ni := int(n)
```

Reply:
I'll make the change. Just to be clear, is this due to the fact that the value passed it might not be an integer?

Reply:
We're implementing Lua 5.2, so yeah, numbers are `float64` internally. It may not be representable as an integer.